csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -58,9 +58,9 @@ public override QueryResult ExecuteQuery()
 
             GetQueryResultsOptions getQueryResultsOptions = new GetQueryResultsOptions();
 
-            if (this.Options != null && this.Options.ContainsKey(BigQueryParameters.GetQueryResultsOptionsTimeoutMinutes))
+            if (this.Options!.TryGetValue(BigQueryParameters.GetQueryResultsOptionsTimeoutMinutes, out string? timeoutMinutes))
```

Comment:
I don't think the null-forgiving operator is what's wanted here. At this line of code, this.Options might well be null in which case this will throw. I think what's wanted instead is `if (this.Options?.TryGetValue(...) == true)`.